reference/docs-conceptual/developer/cmdlet/how-to-invoke-a-pscmdlet-from-within-a-pscmdlet.md

```diff
@@ -6,41 +6,41 @@ title: How to invoke a PSCmdlet from within a PSCmdlet
 ---
 # How to invoke a PSCmdlet from within a PSCmdlet
 
-This example shows how to invoke a **PSCmdlet** from within another **PSCmdlet**. In this example,
-the new PSCmdlet `Get-ClipboardReverse` calls `Get-Clipboard` to get the contents of the clipboard.
-The `Get-ClipboardReverse` reverses the order of the characters and returns the reversed string.
+This example shows how to invoke a script based cmdlet or binary cmdlet inheriting from `PSCmdlet`
+from within a binary cmdlet. In this example, the new cmdlet `Get-ClipboardReverse` calls
+`Get-Clipboard` to get the contents of the clipboard. The `Get-ClipboardReverse` reverses the order
+of the characters and returns the reversed string.
 
 > [!NOTE]
-> The **PSCmdlet** class differs from the **Cmdlet** class. **PSCmdlet** implementations use
+> The `PSCmdlet` class differs from the `Cmdlet` class. `PSCmdlet` implementations use
 > runspace context information so you must invoke another cmdlet using the PowerShell pipeline API.
-> In **Cmdlet** implementations you can call the cmdlet's .NET API directly. For an example, see
+> In `Cmdlet` implementations you can call the cmdlet's .NET API directly. For an example, see
 > [How to invoke a Cmdlet from within a Cmdlet][03].
 
 ## To invoke a cmdlet from within a PSCmdlet
 
-1. Ensure that the assembly that defines the cmdlet to be invoked is referenced and that the
-   appropriate `using` statement is added. In this example, the following namespaces are added.
+1. Ensure that the namespace for the `PowerShell` API is referenced. In this example, the following
+   namespaces are added.
 
     ```csharp
     using System.Management.Automation;   // PowerShell assembly.
     using System.Text;
     ```
 
-1. To invoke a **PSCmdlet** from within another **PSCmdlet** you must use the Pipeline API to
+1. To invoke a command from within another binary cmdlet you must use the `PowerShell` API to
    construct a new pipeline and add the cmdlet to be invoked. Call the
-   [System.Management.Automation.PowerShell.Invoke\<T>()][01] method to invoke the pipeline
+   [System.Management.Automation.PowerShell.Invoke\<T>()][01] method to invoke the pipeline.
 
     ```csharp
-    var ps = PowerShell.Create();
+    using var ps = PowerShell.Create(RunspaceMode.CurrentRunspace);
```

Comment:
The `PowerShell` instance needs to be disposed so I've added `using` here to make that clear (similar to `using (exp) { code }` but handles it after `ps` is no longer in scope).

Also added `RunspaceMode.CurrentRunspace`. Since we know we're on the pipeline thread we can utilize the existing runspace that we're running in. Without this, a new runspace is spun up for every invocation.